app/main.js

```diff
@@ -933,9 +934,18 @@ function createPreferencesWindow () {
 }
 
 function updateTray () {
-  updateToolTip()
-  appIcon.setImage(trayIconPath())
-  appIcon.setContextMenu(getTrayMenu())
+  updateToolTip();
```

Comment:
I think that this should stay as it was before, so those 3 calls: 
```
updateToolTip()
appIcon.setImage(trayIconPath())
appIcon.setContextMenu(getTrayMenu())
```
[appIcon class](https://github.com/hovancik/stretchly/blob/trunk/app/utils/appIcon.js) should stay the only one that is responsible for the icon in tray. We can then expand on [tests](https://github.com/hovancik/stretchly/blob/trunk/test/appIcon.js) as well.

`AppIcon#trayIconFileName` should return correct path to the icon. I would need more time with your code, but it seems to me now that we need to  extend  `constructor ({ platform, paused, monochrome, inverted, darkMode })` with 2 more values, something like: 
- `remainingModeString` - which would be `""`, `"Circle"` or `"Number"`
- `remainingTimeString` - which would be `""`, `"1"`, `"2"` etc...

Then first code will become the second: 

```
      if (this.platform === 'darwin') {
        return `trayMacMonochrome${pausedString}Template.png`
      } else {
        return `trayMonochrome${invertedMonochromeString}${pausedString}.png`
      }
```
-> 

```
      if (this.platform === 'darwin') {
        return `trayMacMonochrome${pausedString}${remainingModeString}${remainingTimeString}Template.png`
      } else {
        return `trayMonochrome${invertedMonochromeString}${pausedString}${remainingModeString}${remainingTimeString}.png`
      }
``` 

`remainingModeString` and  `remainingTimeString` could get calculated by some class based on current state and preferences and return values described up. That is then passed to constructor of `AppIcon`. 

 To go with my example code, let's say we have 2 minutes to break, we are on Mac, monochrome icon mode is on and we wanna show numbers; the resulting file would be: 

> return `trayMacMonochrome${pausedString}${remainingModeString}${remainingTimeString}Template.png`

`trayMacMonochromeNumber2Template.png`

Let's say breaks are paused, we are on Linux, and inverted monochrome, we wanna show circles. All is good and we get correct values as `${remainingModeString}${remainingTimeString}` are both empty strings. 

> `trayMonochrome${invertedMonochromeString}${pausedString}${remainingModeString}${remainingTimeString}.png`

`trayMonochromeInvertedPaused.png`

As mentioned earlier, this can be nicely tested.